CanvasColorSpaceProposal.md

```diff
@@ -26,34 +27,74 @@ Some implementations convert images drawn to canvases to the sRGB color space. T
 
 ## Proposed Solution
 
 * Clearly define the color space of canvases.  Most current browser implementations are either color-managed or are currently actively working on becoming color-managed.  Therefore, it will be possible to specify that the default color space of canvases as 'srgb' instead of leaving it undefined in the spec (as is currently the case due to lack of interoperability).
 * Add a canvas context creation attribute to specify a color space.
-* Add a canvas context creation attribute to specify a numeric format for storing pixel values.
-* Color space and numeric format parameters are also extended to image storage interfaces that interact with canvases, such as CanvasPattern, ImageData and ImageBitmap.
+* Add a canvas context creation attribute to specify an encoding format for storing pixel values.
+* Add a canvas context creation attribute to specify a linearity for perceptual vs physical blending/interpolation.
+* Color space, encoding format, and linearity parameters are also extended to image storage interfaces that interact with canvases, such as CanvasPattern, ImageData and ImageBitmap.
 
 ### Processing Model
 
-#### The colorSpace canvas creation parameter
-
 IDL:
 <pre>
-enum CanvasColorSpace {
+// Feature enums:
+
+enum CanvasColorSpaceEnum {
   "srgb", // default
-  "linear-srgb",
+  "rec-2020",
 };
 
-enum CanvasPixelFormat {
-  "8-8-8-8", // default
+enum CanvasColorEncodingEnum {
+  "unorm8",      // default, 0.5 encoded as 0x80
+  "unorm8-srgb", // 0.5 encoded as 0xbc
   "float16",
 };
 
+enum CanvasColorLinearityEnum {
+  "luminance", // default, 0.5 is half the (perceptual) brightness/luminance of 1.0.
+  "radiance",  // 0.5 is half the (physical) radiance of 1.0. (~75% of the brightness)
+};
+
+// Feature detection:
+
+interface CanvasColorSpace {
+  const CanvasColorSpaceEnum srgb = "srgb";
+  const CanvasColorSpaceEnum rec2020 = "rec-2020";
+};
+
+interface CanvasColorEncoding {
+  const CanvasColorEncodingEnum unorm8 = "unorm8";
+  const CanvasColorEncodingEnum unorm8Srgb = "unorm8-srgb";
+  const CanvasColorEncodingEnum float16 = "float16";
+};
+
+interface CanvasColorLinearity {
+  const CanvasColorLinearityEnum luminance = "luminance";
+  const CanvasColorLinearityEnum radiance = "radiance";
+};
+
+// Feature activation:
+
 partial dictionary CanvasRenderingContext2DSettings {
-  CanvasColorSpace colorSpace = "srgb";
-  CanvasPixelFormat pixelFormat = "8-8-8-8";
+  CanvasColorSpaceEnum colorSpace = "srgb";
+  CanvasColorEncodingEnum colorEncoding = "unorm8";
+  CanvasColorLinearityEnum colorLinearity = "luminance";
+};
+
+partial dictionary WebGLContextAttributes {
+  CanvasColorSpaceEnum colorSpace = "srgb";
+  CanvasColorEncodingEnum colorEncoding = "unorm8";
+  CanvasColorLinearityEnum colorLinearity = "luminance";
+};
+
+dictionary ImageDataColorSettings {
+  CanvasColorSpaceEnum colorSpace = "srgb";
```

Comment:
You should use CanvasColorPrimariesEnum or something similar. Color space is too widely used to mean the whole combination: primaries, transfer function, white point and black point, e.g, in ICC specs.

And since the sRGB primaries are the same as BT.709 you should have an alternate BT.709 enum value that can be used.

Reply:
I believe we could call this "chromaticity", which iirc is `(primaries, white-point)`.

Reply:
Oh, nevermind, this enum is supposed to include transfer function, because that's required for conversions. I removed the old proposal's support of arbitrary `(chromaticity, transfer-func)` pairs.